src/transport/types.ts

```diff
@@ -24,12 +26,21 @@ export interface ITransportEvents {
 }
 export type TransportEventEmitter = StrictEventEmitter<EventEmitter, ITransportEvents>;
 
+export type IPMode = {
+  ip4: boolean;
+  ip6: boolean;
```

Comment:
Can it support both ip4 and ip6? Can it support none? Intuitively seems there are three possible options: ip4, ip4&ip6, ip6. Maybe it's best to have an enum (at least internally) to represent the 3 possible states?